orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/entity/VtrList.java

```diff
@@ -0,0 +1,159 @@
+package uk.gov.di.orchestration.shared.entity;
+
+import com.google.gson.annotations.JsonAdapter;
+import net.minidev.json.JSONArray;
+import net.minidev.json.parser.JSONParser;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import uk.gov.di.orchestration.shared.serialization.VtrListAdapter;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static java.text.MessageFormat.format;
+import static net.minidev.json.parser.JSONParser.DEFAULT_PERMISSIVE_MODE;
+
+@JsonAdapter(VtrListAdapter.class)
+public class VtrList {
+
+    private static final Logger LOG = LogManager.getLogger(VtrList.class);
+
+    public static final VtrList DEFAULT_VTR_LIST = new VtrList(Collections.emptyList());
+
+    private final List<VectorOfTrust> vtr;
+    private final CredentialTrustLevel credentialTrustLevel;
+    private final List<LevelOfConfidence> levelsOfConfidence;
+    private final VectorOfTrust effectiveVectorOfTrust;
+
+    public VtrList(List<VectorOfTrust> vtr) {
+        validateVtr(vtr);
+        this.vtr = !vtr.isEmpty() ? vtr : List.of(VectorOfTrust.DEFAULT_VECTOR_OF_TRUST);
+        this.credentialTrustLevel =
+                this.vtr.stream()
+                        .map(VectorOfTrust::getCredentialTrustLevel)
+                        .min(Comparator.naturalOrder())
+                        .orElseThrow(); // Should never throw.
+        this.levelsOfConfidence =
+                this.vtr.stream()
+                        .filter(vot -> vot.getCredentialTrustLevel() == this.credentialTrustLevel)
+                        .map(VectorOfTrust::getLevelOfConfidence)
+                        .distinct()
+                        .sorted(Comparator.naturalOrder())
+                        .toList();
+        this.effectiveVectorOfTrust =
+                this.vtr.stream()
+                        .filter(
+                                vot ->
+                                        vot.getCredentialTrustLevel()
+                                                .equals(this.credentialTrustLevel))
+                        .filter(vot -> vot.getLevelOfConfidence().equals(levelsOfConfidence.get(0)))
+                        .findFirst()
+                        .orElseThrow(); // Should never throw.
+    }
+
+    public static VtrList of(VectorOfTrust first, VectorOfTrust... rest) {
+        return new VtrList(Stream.concat(Stream.of(first), Arrays.stream(rest)).toList());
+    }
```

Comment:
Do you mean why not just accept a vararg and remove the `first` parameter? Could do, and it would be simpler. Removing the `first` parameter would appear to allow you to create a VtrList with no elements, even though in practice the VTR would have a single default element. I thought given `.of` is generally used in tests, if the author doesn't care about the value of the VtrList they would be better off using `VtrList.DEFAULT_VTR_LIST` rather than `VtrList.Of()`.

Reply:
If you just mean it looks a little convoluted, let me know if there's a better way.  It's a shame there's no `Stream.beginWith`.

Reply:
I think I would expect to just have a vararg as the param. I guess this is probably fine, but I think it's just not super clear why it's being done on first read (ie it's not clear you're doing it so that someone can't call it with no elements)

Reply:
I've added a comment for clarity.